### ISSUE TITLE
Failed to open field term dictionary in composite file. Is the field indexed

### ISSUE THREAD

**Issue by jpastuszek** [tantivy-search/tantivy-cli#26]
I am toying with the CLI. I got my docs indexed but when I try to search with
`RUST_BACKTRACE=1 tantivy search -i test -q 'foo'`

I get this panic:
```
thread 'main' panicked at 'Failed to open field term dictionary in composite file. Is the field indexed', libcore/option.rs:917:5
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
   1: std::panicking::default_hook::{{closure}}
   2: std::panicking::rust_panic_with_hook
   3: std::panicking::begin_panic_fmt
   4: core::panicking::panic_fmt
   5: core::option::expect_failed
   6: tantivy::core::segment_reader::SegmentReader::inverted_index
   7: <core::iter::Map<I, F> as core::iter::iterator::Iterator>::fold
   8: <tantivy::query::term_query::term_query::TermQuery as tantivy::query::query::Query>::weight
   9: <&'a mut I as core::iter::iterator::Iterator>::next
  10: <tantivy::query::boolean_query::boolean_query::BooleanQuery as tantivy::query::query::Query>::weight
  11: tantivy::commands::search::run_search_cli
  12: tantivy::main
  13: std::rt::lang_start::{{closure}}
  14: main
```

My schema looks like this (it has many fields with same options but some are u64 and some are text + one _doc field):
```
    {
      "name": "backend_access.response.headers.Set-Cookie.0",
      "type": "text",
      "options": {
        "indexing": {
          "record": "position",
          "tokenizer": "en_stem"
        },
        "stored": true
      }
    },
    {
      "name": "response.headers.Server.0",
      "type": "text",
      "options": {
        "indexing": {
          "record": "position",
          "tokenizer": "en_stem"
        },
        "stored": true
      }
    },
    {
      "name": "_doc",
      "type": "text",
      "options": {
        "indexing": {
          "record": "position",
          "tokenizer": "en_stem"
        },
        "stored": true
      }
    }
```

**Comment by fulmicoton** [comment]
Can you confirm that you used tantivy 0.5.0 for indexing and searching?
(`tantivy --version`)

Also, can you confirm that you indexed using this schema, and did not modify the schema of the index
after indexing?
If so... It might be a bug. Ideally if you had a reproducing data file that would be super helpful!

**Comment by fulmicoton** [comment]
@jpastuszek Never mind. I was able to reproduce. There is indeed a bug.

**Comment by fulmicoton** [comment]
Can be reproduced if any of the indexed field is missing in a segment.

This is equivalent to, no documents in the segmetn contained the indexed field.

**Comment by fulmicoton** [comment]
Thanks for reporting. This was indeed a bug. It is fixed in version 0.5.1.
